src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -1129,61 +1129,54 @@ public ConnectionState(ConnectionContext connection, HubConnection hubConnection
 
             public void AddInvocation(InvocationRequest irq)
             {
-                lock (_lock)
+                 
+                    
+                if (!_pendingCalls.TryAdd(irq.InvocationId, irq))
                 {
-                    if (_pendingCalls.ContainsKey(irq.InvocationId))
-                    {
-                        Log.InvocationAlreadyInUse(_hubConnection._logger, irq.InvocationId);
-                        throw new InvalidOperationException($"Invocation ID '{irq.InvocationId}' is already in use.");
-                    }
-                    else
-                    {
-                        _pendingCalls.Add(irq.InvocationId, irq);
-                    }
+                    Log.InvocationAlreadyInUse(_hubConnection._logger, irq.InvocationId);
+                    throw new InvalidOperationException($"Invocation ID '{irq.InvocationId}' is already in use.");
                 }
+                     
+                 
             }
 
             public bool TryGetInvocation(string invocationId, out InvocationRequest irq)
             {
-                lock (_lock)
-                {
+                 
                     return _pendingCalls.TryGetValue(invocationId, out irq);
-                }
+                 
             }
 
             public bool TryRemoveInvocation(string invocationId, out InvocationRequest irq)
             {
-                lock (_lock)
-                {
-                    if (_pendingCalls.TryGetValue(invocationId, out irq))
-                    {
-                        _pendingCalls.Remove(invocationId);
-                        return true;
-                    }
-                    else
-                    {
-                        return false;
-                    }
-                }
+                
+
+                    return  _pendingCalls.TryRemove(invocationId, out irq);
+                     
+                
             }
 
             public void CancelOutstandingInvocations(Exception exception)
             {
                 Log.CancelingOutstandingInvocations(_hubConnection._logger);
 
-                lock (_lock)
+
+                //https://msdn.microsoft.com/en-us/library/dd287131.aspx
```

Comment:
Our convention is to put a space after `//` in our comments

Reply:
But hmm, this comment while helpful for review, might not be necessary

Reply:
Should I leave it for now or remove it and added to the PR message ?